jmx/plugins/org.jboss.tools.jmx.jolokia/src/org/jboss/tools/jmx/jolokia/internal/connection/JolokiaMBeanServerConnection.java

```
@@ -246,39 +241,86 @@ public Object invoke(ObjectName name, String operationName, Object[] params, Str
 	public ObjectInstance getObjectInstance(ObjectName name) throws InstanceNotFoundException, IOException {
-		// TODO Auto-generated method stub
-		return null;
+		return createObjectInstance(name);
 	}
 
 	@Override
-	public boolean isInstanceOf(ObjectName name, String className) throws InstanceNotFoundException, IOException {
-		// TODO Auto-generated method stub
-		return false;
-	}
-	@Override
 	public Set<ObjectInstance> queryMBeans(ObjectName name, QueryExp query) throws IOException {
-		// TODO Auto-generated method stub
-		return Collections.emptySet();
+		Set<ObjectInstance> res = new HashSet<>();
+		try {
+			J4pSearchRequest req = new J4pSearchRequest(name.getCanonicalName());
+			J4pResponse<J4pSearchRequest> j4pResponse = j4pClient.execute(req);
+			Object value = j4pResponse.getValue();
+			if(value instanceof JSONArray){
+				for (Object mbean : (JSONArray)value) {
+					if(mbean instanceof String){
+						res.add(createObjectInstance((String)mbean));
+					}
+				}
+			}
+		} catch (MalformedObjectNameException | J4pException e) {
+			Activator.pluginLog().logError(e);
+		}
+		return res;
+	}
+
+	private ObjectInstance createObjectInstance(String mbean) throws MalformedObjectNameException {
+		ObjectName objectName = new ObjectName(mbean);
+		return createObjectInstance(objectName);
+	}
+
+	private ObjectInstance createObjectInstance(ObjectName objectName) {
+		String classname = retrieveClassName(objectName);
+		return new ObjectInstance(objectName, classname);
+	}
+
+	private String retrieveClassName(ObjectName objectName) {
+		J4pListRequest listAttributes = new J4pListRequest(objectName.getDomain()+"/"+objectName.getCanonicalKeyPropertyListString()+"/class");
+		try {
+			J4pResponse<J4pListRequest> listAttributesResponse = j4pClient.execute(listAttributes);
+			return listAttributesResponse.getValue();
+		} catch (J4pException e) {
+			// TODO : can we avoid to catch an exception? How to know if the class attribute need to be called or not?
```

@apupier If you do a query like this you end up at the top-level beans class name. AFAIS, Jolokia [unconditionally add the class name](https://github.com/rhuss/jolokia/blob/82c8a6c41494e40fe14fb67e8b4d73dd31674dbb/agent/core/src/main/java/org/jolokia/handler/list/ClassNameDataUpdater.java#L43) to this info, so I expect as long as MBeanInfo.getClassname() return something its included here.

Of course a `J4pException` can occur for other reasons, too, like network failure. I would raise an error here in this case, at least not silently ignoring it.